enterprise/admin/forms.py

```diff
@@ -280,6 +280,47 @@ def _validate_reason(self):
             raise ValidationError(ValidationMessages.MISSING_REASON)
 
 
+class ManageLearnersDataSharingConsentForm(forms.Form):
+    """
+    Form to request DSC from a learner.
+    """
+    email_or_username = forms.CharField(
+        label=_("Enter an email address or username."),
+        required=True
+    )
+    course = forms.CharField(
+        label=_("Enter the Course key"),
+        required=True,
+    )
+
+    class Fields:
+        """
+        Namespace class for field names.
+        """
+        EMAIL_OR_USERNAME = "email_or_username"
+        COURSE = "course"
+
+    def clean_course(self):
+        """
+        Verify course ID has an associated course in LMS.
+        """
+        course_id = self.cleaned_data[self.Fields.COURSE].strip()
+        client = EnrollmentApiClient()
+        if not client.get_course_details(course_id):
+            raise ValidationError(ValidationMessages.INVALID_COURSE_ID.format(course_id=course_id))
+        return course_id
+
+    def clean_email_or_username(self):
+        """
+        Verify email_or_username has associated user in our database.
+        """
+        email_or_username = self.cleaned_data[self.Fields.EMAIL_OR_USERNAME].strip()
+        email = email_or_username__to__email(email_or_username)
+        if not User.objects.filter(email=email).exists():
```

Comment:
We are only filtering the user here on the basis of email. Then why we are taking the username?

Reply:
`email_or_username__to__email ` function will convert the username to email.